google/resource_cloudfunctions_function.go

```diff
@@ -550,9 +556,30 @@ func flattenEventTrigger(eventTrigger *cloudfunctions.EventTrigger) []map[string
 		return result
 	}
 
+	resource := ""
+	switch {
+	case strings.HasPrefix(eventTrigger.EventType, "google.storage.object."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "google.pubsub.topic."):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "google.firestore.document."):
+		parts := strings.SplitN(eventTrigger.Resource, "/", 6)
+		resource = parts[len(parts)-1]
+		// Legacy style triggers
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.storage/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.pubsub/eventTypes/"):
+		resource = GetResourceNameFromSelfLink(eventTrigger.Resource)
+	case strings.HasPrefix(eventTrigger.EventType, "providers/cloud.firestore/eventTypes/"):
+		parts := strings.SplitN(eventTrigger.Resource, "/", 6)
+		resource = parts[len(parts)-1]
+	default:
```

Comment:
nit: I don't think we'd want to continue having not matched anything., so we should probably omit the default case or return an error if we have one. (both here and above)

Reply:
I have omitted the default case, which was the original behaviour